src/containers/pages/InterventionPlan/IRS/plan/index.tsx

```diff
@@ -1926,11 +2046,54 @@ class IrsPlan extends React.Component<
           : `${Number(newPlan.plan_version) + 1}`,
       };
 
+      const { assignmentsArray, organizationsById } = this.props;
+
       if (newPlanDraft.plan_jurisdictions_ids && newPlanDraft.plan_jurisdictions_ids.length) {
         const planPayload = extractPlanPayloadFromPlanRecord(newPlanDraft);
         if (planPayload) {
-          this.setState({ isSaveDraftDisabled: true }, () => {
-            // todo - handle Finalized plans!!
+          // create payload of assignments based on current store state
+          const nextAssignments: Assignment[] = assignmentsArray.map((a: Assignment) => ({
+            ...a,
+            fromDate: moment(start).format(),
+            toDate: moment(end).format(),
+          }));
+
+          // create temp reference to determine which assignments should be retired on server
+          const assignmentTeamIdsByJurisdictionId: { [key: string]: string[] } = {};
+          for (const assignment of nextAssignments) {
+            if (!assignmentTeamIdsByJurisdictionId[assignment.jurisdiction]) {
+              assignmentTeamIdsByJurisdictionId[assignment.jurisdiction] = [];
+            }
+            assignmentTeamIdsByJurisdictionId[assignment.jurisdiction].push(
+              assignment.organization
+            );
+          }
+
+          // fetch list of assignments from server which need to be retired
+          const retiredAssignments: Assignment[] = await this.getAllAssignments(
+            planPayload.identifier,
+            values(organizationsById),
+            (existingAssignments: Assignment[]) => {
+              const assignmentsToRetire: Assignment[] = [];
+              for (const result of existingAssignments) {
+                if (
+                  !assignmentTeamIdsByJurisdictionId[result.jurisdiction] ||
```

Comment:
not sure what is preferred btwn current and

```javascript
!(result.jurisdiction in assignmentTeamIdsByJurisdictionId) ||
```

Reply:
Oh interesting 🤔

Reply:
Cool looks like it's slightly more performant :)